tasks/src/reinforcement-learning/about.md

```diff
@@ -32,6 +24,72 @@ This place can be filled with variants of this task if there's any.
+
+## Task Variants 
+
+### Q - Learning
+
+* It is used to learn the policy for reinforcement learning
+* Q-function[Q(s,a)]: returns Q-value for action a at state s. Learning an action-value function, a.k.a. Q function, that computes the expected utility of taking an action in a state after training converges.
+* it will find the best course of action, given the current state of the agent. Depending on where the agent is in the environment, it will decide the next action to be taken.
+
+With the help of Q - value and advanced statistics we can find the Deep - Q - Network. You all must have heard of a game called chess. nowadays it is mostly played online, also you don't need a second player to play, you can just play against a bot and also get defeated, which leaves you thinking how can a machine beat me. Well, because it is learning. 
+
+Now, the thing to understand is when training the learner for games like Chess, the agent has a very small action space, but large state space. For example, in the game of chess, the state space is the whole 20X20 board, but the action space is small like a pawn can move only in 1–2 directions. Deep-Q-Network has been proven to be very effective.
+
+But when we talk about reinforcement learning for natural language processing there are few problems which are discussed below:
+
+#### Sequential decision making problem for text understanding
+
+Agent observes state as a string of text at a time 't'. Agent also knows a set of possible actions, each describes as a string text. Agent tries to understand the “state text” and all possible “action texts”, then takes a decision which maximizes the long term reward. Then, the environment state changes to a new state, agent receives an immediate reward.
+
+#### Unbounded action space in reinforcement learning
+
+Not only the state space is huge, but the action space is also huge too. Action is characterized by unbounded natural language descriptions. For example, if say to the model “Hi! how are you? Have you eaten your lunch? Don't forget to call me tomorrow as it is the bigday ”. Well, this input text from me is the state-space for the model(quite heavy), and the action space is every text combination available(or infinity). This problem for such a huge action space was still the problem in Deep-Q-Network.
+
+For solving this problem DRRN(deep reinforcement relevance network) was proposed. The idea of DRRN is to project both the state and action into a continuous space(as vectors). Q-function is a relevance function of the state vector and action vector.the DRRN experiences higher average reward.

```

Comment:
I see it's taken from here, again, we really don't need academic content. We want beginners to understand too! https://paperswithcode.com/paper/deep-reinforcement-learning-with-a-natural/review/